ufl/algorithms/formsplitter.py

```diff
@@ -147,16 +155,16 @@ def extract_blocks(form, i: Optional[int] = None, j: Optional[None] = None):
         if arity > 1:
             for pj in range(num_parts):
                 f = fs.split(form, pi, pj)
-                if f.empty():
+                # Ignore empty forms and linear forms
+                if f.empty() or len(f.arguments()) != 2:
```

Comment:
Do we ever get `arity > 1` and  `len(f.arguments()) != 2` ? Should we (at least) raise a Warning in that case, just to make sure we don't have a term in the form that has not been taken into account (might be due to user mistake, but as it is the user won't know)

Reply:
Yes, formsplitter can return a linear or functional at this point.